### PR TITLE
Removed outdated language referencing "Private Relelase, Teams channe…

### PR DESCRIPTION
**Pull request by jonvsm**
…l, etc

I filed this issue https://github.com/radius-project/docs/issues/816, then made chanages directly in the file to address it

Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### Summary
🌐🤝🚀

Improved the community overview page in `docs/content/community/overview.md` by removing outdated content and adding new links to the project's GitHub and Discord platforms.

> _The community page was a mess_
> _With outdated and redundant text_
> _So they simplified it_
> _And added a bit_
> _Of `radius-project` and Discord, no less_

### Walkthrough
*  Simplify and update the introduction paragraph to reflect the open status of the Radius community ([link](https://github.com/radius-project/docs/pull/817/files?diff=unified&w=0#diff-e3770355c61f4e617f6541a7674d1d265d15868ffec8a8a852324e86b5716fc7L9-R13))
* Remove the sentence about contacting the Radius Core Team for meeting invites, as it is no longer necessary ([link](https://github.com/radius-project/docs/pull/817/files?diff=unified&w=0#diff-e3770355c61f4e617f6541a7674d1d265d15868ffec8a8a852324e86b5716fc7L20-R24))
* Update the paragraph about GitHub access to include the link to the radius-project organization and the use of Codespaces ([link](https://github.com/radius-project/docs/pull/817/files?diff=unified&w=0#diff-e3770355c61f4e617f6541a7674d1d265d15868ffec8a8a852324e86b5716fc7L45-R49))
* Replace the Teams channel with the Discord server as the preferred way of communication for getting involved ([link](https://github.com/radius-project/docs/pull/817/files?diff=unified&w=0#diff-e3770355c61f4e617f6541a7674d1d265d15868ffec8a8a852324e86b5716fc7L45-R49))

### CODE REVIEW

File: docs/content/community/overview.md [hunk 1]
```diff
@@ -6,22 +6,22 @@ description: "Information about the Radius community"
 weight: 80
 ---
 
-Welcome to the Radius community. We are currently in a private release phase which is open to Microsoft FTEs and select partners. The team is working on a public preview release which will be announced on this page.
+Welcome to the Radius community!
 
 ## Discussions
 
-The best way to interact with us currently is through Discord over at the Radius Discord server:
+The best way to interact with the community if via the Radius Discord server:
```
Comment (willtsai) [inline docs/content/community/overview.md]:
```suggestion
The best way to interact with the community is via the Radius Discord server:
```

File: docs/content/community/overview.md [hunk 2]
```diff
@@ -42,8 +42,8 @@ To present a specific topic on the next Radius community call start by filling o
 
 ## GitHub Issues & repositories
 
-GitHub access is optional and not required to try out and run Radius. If you would like to directly file Issues and have access to the source code, please contact your Radius Core Team contact.
+GitHub access is optional and not required to try out and run Radius. If you would like to directly file Issues, have access to source code, or use Codespaces please join https://github.com/radius-project.
```
Comment (willtsai) [inline docs/content/community/overview.md]:
```suggestion
If you would like to file Issues, access the source code, or use Codespaces please visit the [Radius GitHub repo] (https://github.com/radius-project).
```

File: docs/content/community/overview.md [hunk 3]
```diff
@@ -42,8 +42,8 @@ To present a specific topic on the next Radius community call start by filling o
...
 
 ## How can I get involved?
 
-One of the easiest ways to contribute is to participate in discussions at community engagements or the Teams channel. We are always seeking feedback from. Suggestions around improving the product and docs are always welcome.
+One of the easiest ways to contribute is to participate in discussions at community engagements or via the Discord server referenced above. We are always seeking feedback from. Suggestions around improving the product and docs are always welcome.
```
Comment (willtsai) [inline docs/content/community/overview.md]:
```suggestion
One of the easiest ways to contribute is to participate in discussions at community engagements or via the Discord server referenced above. We are always seeking feedback, especially for suggestions around improving the product and docs.
```

**Review by willtsai — DISMISSED** [review]
will we hold off on merging in this PR until the repo is publicly released in open source?

**Review by AaronCrawfis — APPROVED** [review]

**Comment by AaronCrawfis** [comment]
> will we hold off on merging in this PR until the repo is publicly released in open source?

No need to wait. We don't enter an "invalid" state by merging this as nothing is incorrect.

**Pull request merged**